ignition_ros2_control/src/ignition_system.cpp

```diff
@@ -0,0 +1,402 @@
+      this->dataPtr->sim_joints_[j]);
+
+    // TODO(ahcorde): Revisit this part
+    // Get the joint force
+    // const auto * jointForce =
+    //   this->dataPtr->ecm->Component<ignition::gazebo::components::JointForce>(
+    //   this->dataPtr->sim_joints_[j]);
+
+    // Get the joint position
+    const auto * jointPositions =
+      this->dataPtr->ecm->Component<ignition::gazebo::components::JointPosition>(
+      this->dataPtr->sim_joints_[j]);
+
+    this->dataPtr->joint_position_[j] = jointPositions->Data()[0];
+    this->dataPtr->joint_velocity_[j] = jointVelocity->Data()[0];
+    // this->dataPtr->joint_effort_[j] = jointForce->Data()[0];
+  }
+  return hardware_interface::return_type::OK;
+}
+
+hardware_interface::return_type IgnitionSystem::write()
+{
+  for (unsigned int j = 0; j < this->dataPtr->joint_names_.size(); j++) {
+    if (this->dataPtr->joint_control_methods_[j] & VELOCITY) {
+      if (!this->dataPtr->ecm->Component<ignition::gazebo::components::JointVelocityCmd>(
+          this->dataPtr->sim_joints_[j]))
+      {
+        this->dataPtr->ecm->CreateComponent(
+          this->dataPtr->sim_joints_[j],
+          ignition::gazebo::components::JointVelocityCmd({0}));
+      } else {
+        const auto jointVelCmd =
+          this->dataPtr->ecm->Component<ignition::gazebo::components::JointVelocityCmd>(
+          this->dataPtr->sim_joints_[j]);
+        *jointVelCmd = ignition::gazebo::components::JointVelocityCmd(
+          {this->dataPtr->joint_velocity_cmd_[j]});
+      }
+    }
+
+    if (this->dataPtr->joint_control_methods_[j] & POSITION) {
+      if (!this->dataPtr->ecm->Component<ignition::gazebo::components::JointPositionReset>(
+          this->dataPtr->sim_joints_[j]))
+      {
+        this->dataPtr->ecm->CreateComponent(
+          this->dataPtr->sim_joints_[j],
+          ignition::gazebo::components::JointPositionReset({this->dataPtr->joint_position_[j]}));
+        const auto jointPosCmd =
+          this->dataPtr->ecm->Component<ignition::gazebo::components::JointPositionReset>(
+          this->dataPtr->sim_joints_[j]);
+        *jointPosCmd = ignition::gazebo::components::JointPositionReset(
+          {this->dataPtr->joint_position_cmd_[j]});
+      }
+    }
+
+    if (this->dataPtr->joint_control_methods_[j] & EFFORT) {
+      if (!this->dataPtr->ecm->Component<ignition::gazebo::components::JointForceCmd>(
+          this->dataPtr->sim_joints_[j]))
+      {
+        this->dataPtr->ecm->CreateComponent(
+          this->dataPtr->sim_joints_[j],
+          ignition::gazebo::components::JointForceCmd({0}));
+      } else {
+        const auto jointEffortCmd =
+          this->dataPtr->ecm->Component<ignition::gazebo::components::JointForceCmd>(
+          this->dataPtr->sim_joints_[j]);
+        *jointEffortCmd = ignition::gazebo::components::JointForceCmd(
+          {this->dataPtr->joint_effort_cmd_[j]});
+      }
+    }
+  }
+
+  // // Get the simulation time and period
```

Comment:
This information is provided by `UpdateInfo` on Ignition's update callbacks, in case you still need it